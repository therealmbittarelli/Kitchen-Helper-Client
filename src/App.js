import React from "react";
import { Route, Switch } from "react-router-dom";
import LoginRoute from "./Routes/LoginRoute";
import RegistrationRoute from "./Routes/RegistrationRoute";
import DashboardRoute from "./Routes/DashboardRoute";
import PrivateRoute from "./Components/PrivateOnly/PrivateRoute";
import PublicOnlyRoute from "./Components/PublicOnly/PublicOnlyRoute";
import RecipesRoute from "./Routes/RecipesRoute";
import PantryRoute from "./Routes/PantryRoute";
import PlannerRoute from "./Routes/PlannerRoute";
import MealPlan from "./Components/Planner/MealPlan";
import NotFoundRoute from "./Routes/NotFoundRoute";
import Darkmode from "darkmode-js";
import Header from "./Components/Header/Header";
import Landing from "./Routes/LandingPage/LandingPage";
import searchRecipes from "./Components/Recipes/Search-Recipe";
import individualRecipe from "./Components/Recipes/Individual-Recipe";
import editRecipe from "./Components/Recipes/Edit-Recipe";
import createRecipes from "./Components/Recipes/Create-Recipe";
import viewSearchRecipes from "./Components/Recipes/View-Search";

const options = {
  bottom: "64px", // default: '32px'
  right: "32px",
  left: "unset",
  time: "0.5s", // default: '0.3s'
  mixColor: "#fff", // default: '#fff'
  backgroundColor: "#fff", // default: '#fff'
  buttonColorDark: "#100f2c", // default: '#100f2c'
  buttonColorLight: "#fff", // default: '#fff'
  saveInCookies: false, // default: true,
  label: "ðŸŒ“", // default: ''
  autoMatchOsTheme: true // default: true
};

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      currentUser: {},
      isLoggedIn: false,
      recipes: [],
      hasError: false
    };
  }

  render() {
    const darkmode = new Darkmode(options);
    darkmode.showWidget();

    return (
      <div className="App">
        <main>
          <Header user={this.state.currentUser} />
          <Switch>
            <PrivateRoute
              exact
              path={"/home"}
              component={DashboardRoute}
            />
            <PrivateRoute
              exact
              path={"/recipes"}
              component={RecipesRoute}
            />
            <PrivateRoute
              exact
              path={"/recipes/search"}
              component={searchRecipes}
            />
            <PrivateRoute
              exact
              path={"/recipes/create"}
              component={createRecipes}
            />
            <PrivateRoute
              exact
              path={"/recipes/:recipeId"}
              component={individualRecipe}
            />
            <PrivateRoute
              exact
              path={"/edit-recipe/:recipeId"}
              component={editRecipe}
            />
            <PrivateRoute
              exact
              path={"/recipes/search/:recipeId"}
              component={viewSearchRecipes}
            />
            <PrivateRoute
              exact
              path={"/pantry"}
              component={PantryRoute}
            />
            <PrivateRoute
              exact
              path={"/planner"}
              component={PlannerRoute}
            />
            <PrivateRoute
              exact
              path={"/planner/:mealPlanId"}
              component={MealPlan}
            />
            <PublicOnlyRoute
              exact
              path={"/"}
              component={Landing}
            />
            <PublicOnlyRoute
              exact
              path={"/register"}
              component={RegistrationRoute}
            />
            <PublicOnlyRoute
              exact
              path={"/login"}
              component={LoginRoute}
            />
            <Route
              component={NotFoundRoute}
            />
          </Switch>
        </main>
      </div>
    );
  }
}

export default App;
