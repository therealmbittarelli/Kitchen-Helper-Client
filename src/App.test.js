import React from "react";
import ReactDOM from "react-dom";
import App from "./App";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";

describe("App", () => {
  it("renders without crashing", () => {
    Object.defineProperty(window, "matchMedia", {
      writable: true,
      value: jest.fn().mockImplementation(query => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn()
      }))
    });
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <App />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
