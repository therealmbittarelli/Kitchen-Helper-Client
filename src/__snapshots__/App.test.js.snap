// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the UI as expected 1`] = `
<div
  className="App"
>
  <main>
    <header
      className="heading"
    >
      <div
        className="landing-image"
      >
        <a
          href="http://localhost:3000/"
        >
          <img
            alt="Landing Logo"
            className="landinglogo"
            src="LandingLogo.gif"
          />
        </a>
        <a
          className="login-page-link"
          href="/"
          onClick={[Function]}
        />
      </div>
      <nav
        className="login-page-heading"
      >
        <a
          className="login-page-login login-page-link"
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
         
        <a
          className="login-page-register login-page-link"
          href="/register"
          onClick={[Function]}
        >
          Sign up
        </a>
      </nav>
    </header>
    <div
      className="Landing"
    >
      <header
        className="flex-container"
        id="Landing-Header"
      >
        <div
          className="flex-item branding"
        />
        <div
          className="flex-item"
        />
      </header>
      <section
        id="landingDocumentation"
      >
        <div
          className="site-description"
        >
          <p
            className="p1"
          >
            Kitchen Helper is a best friend in the kitchen. This app combines the ability to track the items in your pantry, the opportunity to plan your meals for a given time period and puts a large database of recipes at your fingertips. You also have the ability to add your own recipes and create shopping lists once your meals are planned.
          </p>
          <p
            className="p2"
          >
            Kitchen Helper will be a time and money saver, helping you create amazing meals at home while staying organized and never running out of that special ingredient you need for your amazing lasagna recipe.
          </p>
          <div
            className="login-signup"
          >
            <div
              className="LoginBtn"
            >
              <label
                className="login-label"
              >
                Login: 
              </label>
              <a
                className="login-btn"
                href="/login"
                onClick={[Function]}
              >
                Let's Get Cooking!
              </a>
            </div>
            <div
              className="RegistrationBtn"
            >
              <label
                className="registration-label"
              >
                Sign Up:
              </label>
              <a
                className="registration-btn"
                href="/register"
                onClick={[Function]}
              >
                Come Join the Fun!
              </a>
            </div>
          </div>
          <h4>
            Please test out our site using the username of 'Guestuser' and the password of !1asdf@@@M
          </h4>
          <div
            className="sampleImages"
          >
            <img
              alt="Pantry Landing"
              className="landingSamples"
              id="recipe-search"
              src="pantryLanding.gif"
            />
            <img
              alt="Home Page"
              className="landingSamples"
              id="pantry-add-form"
              src="homePage.gif"
            />
            <img
              alt="Recipe Detail"
              className="landingSamples"
              id="meal-planning-add-button"
              src="recipeDetail.gif"
            />
          </div>
        </div>
      </section>
      <section
        id="Landing-Buttons"
      >
        <hr
          id="Connect-Divider"
        />
        <div
          id="buttons"
        />
      </section>
    </div>
  </main>
</div>
`;
